platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/InputWire.java

```diff
@@ -0,0 +1,172 @@
+     * and path of least resistance is to just cast to the proper data type.
+     *
+     * <p>
+     * Warning: this will appease the compiler, but it is possible to cast a wire into a data type that will cause
+     * runtime exceptions. Use with appropriate caution.
+     *
+     * @param <A> the input type to cast to
+     * @param <B> the output type to cast to
+     * @return this, cast into whatever type is requested
+     */
+    @NonNull
+    @SuppressWarnings("unchecked")
+    public final <A, B> InputWire<A, B> cast() {
+        return (InputWire<A, B>) this;
+    }
+
+    /**
+     * Convenience method for creating an input wire with a specific input type. This method is useful for when the
+     * compiler can't figure out the generic type of the input wire. This method is a no op.
+     *
+     * @param inputType the input type of the input wire
+     * @param <X>       the input type of the input wire
+     * @return this
+     */
+    @SuppressWarnings("unchecked")
+    public <X> InputWire<X, O> withInputType(@NonNull final Class<X> inputType) {
+        return (InputWire<X, O>) this;
+    }
+
+    /**
+     * Bind this input wire to a handler. A handler must be bound to this inserter prior to inserting data via any
+     * method.
+     *
+     * @param handler the handler to bind to this input wire
+     * @return this
+     * @throws IllegalStateException if a handler is already bound and this method is called a second time
+     */
+    @SuppressWarnings("unchecked")
+    @NonNull
+    public InputWire<I, O> bind(@NonNull final Consumer<I> handler) {
+        if (this.handler != null) {
+            throw new IllegalStateException("Handler already bound");
+        }
+        this.handler = (Consumer<Object>) Objects.requireNonNull(handler);
+
+        return this;
+    }
+
+    /**
+     * Bind this input wire to a handler. A handler must be bound to this inserter prior to inserting data via any
+     * method.
+     *
+     * @param handler the handler to bind to this input task scheduler
+     * @return this
+     * @throws IllegalStateException if a handler is already bound and this method is called a second time
+     */
+    @SuppressWarnings("unchecked")
+    @NonNull
+    public InputWire<I, O> bind(@NonNull final Function<I, O> handler) {
+        if (this.handler != null) {
+            throw new IllegalStateException("Handler already bound");
+        }
+        this.handler = i -> {
+            final O output = handler.apply((I) i);
+            if (output != null) {
+                taskScheduler.forward(output);
+            }
+        };
+
+        return this;
+    }
+
+    /**
+     * Add a task to the task scheduler. May block if back pressure is enabled.
+     *
+     * @param data the data to be processed by the task scheduler
+     */
+    public void put(@Nullable final I data) {
+        taskScheduler.put(handler, data);
```

Comment:
it would be nice to throw if the input wire is unbound when this is invoked.

```
throwIfUnbound();
taskScheduler.put(handler, data);
```

Reply:
I intentionally didn't add nullity checks like this to the framework due to the performance critical nature it will fill. If you aren't convinced, perhaps we can discuss.